src/main/java/com/flightstats/hub/app/HubProperties.java

```diff
@@ -117,9 +153,21 @@ public static Properties loadProperties(String fileName) throws MalformedURLExce
             properties = loadProperties(new File(fileName).toURI().toURL(), true);
         }
         HubProperties.setProperties(properties);
+        ensureSubsystemsDisabledIfReadOnly();
         return properties;
     }
 
+    private static void ensureSubsystemsDisabledIfReadOnly() {
+        if (isReadOnly()) {
```

Comment:
Bear in mind my lack of experience in Java patterns, but my personal feeling is that I'd prefer to iterate over the properties and log an error for each one that is not set to required value before forcing a shutdown as opposed to force setting properties during runtime.

Reply:
I think this is being called at startup, so this is ensuring a read-only version starts with these as false?

Reply:
The hope here was to keep the props external for r/w nodes (say if you want to attach a local write node w/o s3verifier/webhook leadership) but avoid forcing r/o nodes to always set six extra props to false.